lib/iris/_concatenate.py

```diff
@@ -646,22 +789,17 @@ def _cmp(coord, other):
 
         """
         # A candidate axis must have non-identical coordinate points.
-        candidate_axis = not array_equal(coord.core_points(), other.core_points())
+        candidate_axis = not array_equal(coord.points, other.points)
```

Comment:
These are `DimCoord`s so the points and bounds are already numpy arrays. Using `points` instead of `core_points()` makes it more clear to the reader of the code that performing this comparison multiple times will only have a small impact on computational performance, as it does not load data from disk multiple times over.